cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -291,6 +310,19 @@ public void addOutput(String datasetName, Map<String, String> arguments) {
 
   @Override
   public void addOutput(String outputName, OutputFormatProvider outputFormatProvider) {
+    try {
```

Comment:
If you suffix sources with -input and sinks with -output, then how will the lineage be tracked if one pipeline writes to sink "xyz" and another pipeline reads from source "xyz"? Isn't the intention that this sink and source will be represented by the same external dataset? If you use different suffixes, then they will not.